### ISSUE TITLE
Recordings Q1 2023 Megaissue

### ISSUE THREAD

**Issue by benjackwhite** [PostHog/posthog#13393]
# Recordings Q1 2023 Megaissue

In the spirit of our [previous Megaissue](https://github.com/PostHog/product-internal/issues/354), here is the one for Q1 2023. It is designed to be a focus point to link together what we plan to do, feedback on that plan and progress as we go forward.

It is **not** meant to be discussion about individual features. As soon as the discussion starts a dedicated feature issue should be created, added to the roadmap and discussed there.

# OKRs

### ✅ KR1: Release iOS Mobile Recordings to 5 happy beta testers
> **Why?** This is the most requested feature for recordings according to customer success. There is a whole market (Mobile B2C) we currently can’t address because we don’t support this functionality. We’ll start with just iOS, then use this to extend to React Native. Learnings from this should make following up with Android straightforward.

### ✅ KR2: Release the ‘Network Tab’ to 10 happy customers
> **Why?** This will allow our target users, product engineers, to debug problematic sessions more quickly and therefore get more value out of the product

### ✅  KR: Rollout RRWeb2 and improve our ability to debug / improve recording issues
> Why? We are heavy users of RRweb but we aren't super active in the improvement / maintenance of it. If we can test, debug and improve this part of the product we can react to incoming customers with the kind of iterative response they expect

### ✅  KR: Increase data retention for paying users to > 3 months
> Why? One of our USPs is that you can dive deeper on your Insight by looking at recordings related to a funnel, insight etc. Currently we are limited by our storage method but we have a plan to use cheaper, long term storage so we should execute on it.

## Side Quests
In addition to the OKRs we have a couple of side quests of things that are either partially in-progress or would be good to tackle if it doesn't pull us too far from our goals.
- [x] **Increase Recording retention** - For paying users, have a way to increase retention to a sensible amount.
- [x] **Upgrade to RRWeb 2** - Has a lot of fixes and upgrades (like iframe recording) but is currently in an alpha state

# Roadmap of features for the coming months

> Below are headlines features with some notes regarding their implementation. If these are to be worked on / fleshed out further we should create a dedicated issue and link it here. They are **roughly prioritized top to bottom** and are an indication of our intent based on our OKRs. As always - **keep scrappy** - if we realize something here doesn't make sense or the priority should be changed, then go for it.

### ⚡️ Network / Performance panel
**Why:** We have PostHog Events and Console Logs - the next step is Network recording. With this, user's should have the fullest picture of what was going on for a user's experience, which API requests were slow etc.

**How:**
  * [x] MVP the web performance gathering and display - https://github.com/PostHog/posthog/pull/13299
  * [x] Update the Inspector area to be more "slick" with a unified view of all events, better filtering options and a better UX https://github.com/PostHog/posthog/issues/13374

### 📱 Mobile Session Recordings (iOS first)
**Why:** There are many recording tools out there for web and different tools for mobile but few for both. If we could unlock mobile session recordings, even with reduced functionality, it could be huge - especially for self-hosted customers who have few options in this space.

**How:**
  * [x] Run a PoC implementation for something like React Native to prove the idea ([wip here](https://github.com/PostHog/posthog-js-lite/tree/feat/session-recording-experiment)
  * [x] Focus on iOS first (with RN iOS an easy follow up) with the bare minimum of features - recording, masking text inputs
  * [ ] Get feedback from initial users and follow up with things like custom filtering
  * [x] "Wireframe mode" - use image processing to drastically reduce the info captured - great for those with highly sensitive data

### Increased storage retention
**Why:** This is both a current infrastructure issue as well as a common customer request for increased Recordings retention as many recordings are only useful weeks later when delving deeper to an insight

**How:**
- [ ] Implement new ingestion / storage flow in parallel with existing one [see this RFC](https://github.com/PostHog/product-internal/issues/357)

### Upgrade to RRWeb 2 

**Why:** Has a lot of fixes and upgrades (like iframe recording) but is currently in an alpha state

**How**:
- [x] https://github.com/PostHog/posthog-js/issues/523
- [ ] https://github.com/PostHog/posthog/issues/14053

----

### Other work
- [ ] https://github.com/PostHog/posthog/issues/13392
- [x] https://github.com/PostHog/posthog/issues/13948
- [ ] https://github.com/PostHog/posthog/issues/13184

## Ideation for Q2

- [ ] @alexkim205 - many requests from customers for request and response payloads
- [ ] concrete steps / plan to rollout RRWeb2
- [ ] Why do we want to do ingestion rewrite now - ingestion issues are crippling us
  - [ ] Can / should we do this with pipeline team

- [ ] Big bet - Notebooks
  - why? Feeds into 3000 
  - Combine project-wise with Michael
- [ ] Android SDK - contract out? 
- [ ] Lock down recordings as a product (reliability > features)

- [ ] Q3 - Q4 : Team Monitoring 
  - why? Expansion to cover remaining core areas of dev exp - exceptions, alerting
  - why not now? - Lacking the right team expertise and would need some considerable growth
